bokehjs/src/lib/models/axes/axis.ts

```diff
@@ -179,17 +199,40 @@ export class AxisView extends GuideRendererView {
 
     const [nx, ny] = this.normals
     const standoff = extents.tick + extents.tick_label + this.model.axis_label_standoff
-
     const {vertical_align, align} = this.panel.get_label_text_heuristics("parallel")
-    axis_label.position = {
+    const angle = this.panel.get_label_angle_heuristic("parallel")
+
+    const position = {
       sx: sx + nx*standoff,
       sy: sy + ny*standoff,
       x_anchor: align,
       y_anchor: vertical_align,
     }
-    axis_label.align = align
 
-    axis_label.paint(ctx)
+    const axis_label = text instanceof MathText
+      ? this.axis_label_math_text_view
+      : new TextBox({text})
+
+    axis_label.angle = angle
+    axis_label.visuals = this.visuals.axis_label_text
+    axis_label.position = position
+
+    if (axis_label instanceof MathTextView) {
+      // If the image is not loaded
+      if (!axis_label.svg_image) {
```

Comment:
When writing comments like this, you are effectively telling not much more, if anything at all, than the code already does or should have told. Looking at this clause, I can image than the question asked is whether the image was loaded or not. Perhaps a better approach would be to provide a getter with a more meaningful name (e.g. `has_image_loaded`). This would be also better, because you wouldn't expose `svg_image`, which to me is internal to `MathTextView`  and shouldn't be exposed (ask yourself do we care that rendering is done by using SVG?).

Make your code do the talking. Afterall, the code is what we can actually verify by type-checking or running it. Whereas comments are just text. Often redundant text, that later won't get updated, causing more confusion. Write comments where it matters:

1. The code is inherently meaningless to the naked eye and requires explanation what it actually does. For example there is a need for a micro-optimization, which would make the code hard to understand. In that case, sure, let the reader know what is actually going on.
2. You need to explain why something is done that way it was done. For example, you have to `some_var as any` in some context. This begs for an explanation why this is there. Maybe the typings are bad? If so, which specifically. When this could be removed, e.g. point to an related issue or something?